### ISSUE TITLE
Correct way to initialize SFM with an existing model

### ISSUE THREAD

**Issue by meder411** [colmap/colmap#658]
I have been initializing my reconstruction by first creating the model files without points, [as described in the FAQs](https://colmap.github.io/faq.html#reconstruct-sparse-dense-model-from-known-camera-poses), but I am not sure exactly how to proceed from there.

The FAQs say to next run `colmap point_triangulator` and `colmap bundle_adjuster` as needed. 

However, there are also the instructions regarding [registering new images to an existing model](https://colmap.github.io/faq.html#register-localize-new-images-into-an-existing-reconstruction). In that example, it seems one can use `colmap mapper` and provide the containing directory of the model as `--input_path`. 

I am not sure, though, if running the `mapper` with the `--input_path` option after initializing my model by creating the 3 model files actually works. My reason for suspicion is that I also set the option `Mapper.ba_refine_focal_length: 1` to be `0`, and yet the focal length is still refined.

The docs make it seem like modifying the database directly is only necessary when you want to leverage intrinsics for feature matching in highly distorted images. Is directly modifying the DB also necessary to preserve focal lengths?

In summary, **what are the exact steps to initialize a model?** 

1. Can I run `mapper` with the `--input_path` instead of running `point_triangulator` and `bundle_adjustor`? Will this still work?
2. Do I need to write the cameras and poses to the DB directly? If not, how do I prevent COLMAP from trying to refine the focal lengths?

**Comment by ahojnnes** [comment]
You can only continue a reconstruction that has 3D points using the `mapper`. In your case, if you just want to triangulate a scene from existing camera poses, then `point_triangulator` and `bundle_adjuster` should be the right choice. Does that help? Not sure I understand your question correctly.

**Comment by meder411** [comment]
Ah, okay, this does clarify things a bit.

It sounds like after running `point_triangulator` and `bundle_adjuster`, I could always call the `mapper` for subsequent registrations.

Does `point_triangulator` + `bundle_adjuster` = `mapper`? That is, is the `mapper` simply just calling them each once sequentially? I was under the impression that the mapper iteratively called each component.

**Comment by tsattler** [comment]
@meder411 : COLMAP does not update the camera parameters in the database but rather uses them as priors (see also issue #654). As such, only the output model contains the updated camera parameters.

**Comment by meder411** [comment]
That clears things up immensely, thank you!

**Comment by MyLovelyAxe** [comment]
> You can only continue a reconstruction that has 3D points using the `mapper`. In your case, if you just want to triangulate a scene from existing camera poses, then `point_triangulator` and `bundle_adjuster` should be the right choice. Does that help? Not sure I understand your question correctly.

Hi, if I have known camera poses available but I also know they are not completely precise, I want to let `bundle_adjuster` to refine the known camera poses based on generated images.bin, cameras.bin and points3D.bin, is this possible?
By the way, may I ask which python function does the work of `colmap bundle_adjuster`? I have tried `pycolmap.bundle_adjustment` defined in [colmap/src/pycolmap/pipeline/sfm.cc](https://github.com/colmap/colmap/blob/c238aec0e669610850badf3a3279dc2858f37f0f/src/pycolmap/pipeline/sfm.cc#L129), but this function neither returns an adjusted reconstruction nor edits any files saved on hard disk.
